File: hackathon_CPP_SDK/src-gen/nevonex/fcal/GPS_INFO_R.cpp
/*
 * nevonex/fcal/GPS_INFO_R.cpp
 * Copyright (c) Robert Bosch GmbH. All rights reserved.
 */

#include "GPS_INFO_R.hpp"
#include <ecore/EObject.hpp>
#include <ecore/EClass.hpp>
#include "nevonex/fcal/FcalPackage.hpp"
#include <ecorecpp/mapping.hpp>
#include <algorithm>
#include <nevonex/fcb/SubscriberEnum.hpp>
#include <nevonex-fcal-platform/log/Logger.hpp>

#if BOOST_VERSION >= 106501
#define BOOST_STACKTRACE_HEADERS_FOUND
#include <boost/stacktrace.hpp>
#endif

/*PROTECTED REGION ID(GPS_INFO_R.cpp) START*/
// Please, enable the protected region if you add manually written code.
// To do this, add the keyword ENABLED before START.
/*PROTECTED REGION END*/

using namespace ::nevonex::fcal;
using namespace ::nevonex::log;

// Default constructor
GPS_INFO_R::GPS_INFO_R() : m_latitude(0.0), m_longitude(0.0), m_altitude(0.0), m_positionTime(0.0), m_hDOP(0.0f), m_pDOP(0.0f), m_numberOfSattelites(0)
{

    /*PROTECTED REGION ID(GPS_INFO_R__GPS_INFO_R) START*/
    // Please, enable the protected region if you add manually written code.
    // To do this, add the keyword ENABLED before START.
    /*PROTECTED REGION END*/

}

GPS_INFO_R::~GPS_INFO_R()
{

    /*PROTECTED REGION ID(GPS_INFO_R__Destructor) START*/
    // Please, enable the protected region if you add manually written code.
    // To do this, add the keyword ENABLED before START.
    /*PROTECTED REGION END*/

}

// Attributes

::ecore::EDouble GPS_INFO_R::getLatitude() const
{
    return m_latitude;
}

void GPS_INFO_R::setLatitude(::ecore::EDouble _latitude)
{

    ::ecore::EDouble _old_latitude = m_latitude;
    m_latitude = _latitude;

}

::ecore::EDouble GPS_INFO_R::getLongitude() const
{
    return m_longitude;
}

void GPS_INFO_R::setLongitude(::ecore::EDouble _longitude)
{

    ::ecore::EDouble _old_longitude = m_longitude;
    m_longitude = _longitude;

}

::ecore::EDouble GPS_INFO_R::getAltitude() const
{
    return m_altitude;
}

void GPS_INFO_R::setAltitude(::ecore::EDouble _altitude)
{

    ::ecore::EDouble _old_altitude = m_altitude;
    m_altitude = _altitude;

}

::ecore::EDouble GPS_INFO_R::getPositionTime() const
{
    return m_positionTime;
}

void GPS_INFO_R::setPositionTime(::ecore::EDouble _positionTime)
{

    ::ecore::EDouble _old_positionTime = m_positionTime;
    m_positionTime = _positionTime;

}

::ecore::EFloat GPS_INFO_R::getHDOP() const
{
    return m_hDOP;
}

void GPS_INFO_R::setHDOP(::ecore::EFloat _hDOP)
{

    ::ecore::EFloat _old_hDOP = m_hDOP;
    m_hDOP = _hDOP;

}

::ecore::EFloat GPS_INFO_R::getPDOP() const
{
    return m_pDOP;
}

void GPS_INFO_R::setPDOP(::ecore::EFloat _pDOP)
{

    ::ecore::EFloat _old_pDOP = m_pDOP;
    m_pDOP = _pDOP;

}

::ecore::EInt GPS_INFO_R::getNumberOfSattelites() const
{
    return m_numberOfSattelites;
}

void GPS_INFO_R::setNumberOfSattelites(::ecore::EInt _numberOfSattelites)
{

    ::ecore::EInt _old_numberOfSattelites = m_numberOfSattelites;
    m_numberOfSattelites = _numberOfSattelites;

}

// References

File: com.bosch.fsp.hackathon.gen/src/com/bosch/nevonex/fcal/impl/GPS_INFO_R.java
/**
Copyright (c) Robert Bosch GmbH. All rights reserved.
*/
package com.bosch.nevonex.fcal.impl;

import com.bosch.nevonex.fcal.IGPS_INFO_R;

import org.eclipse.emf.ecore.EClass;
import org.eclipse.emf.ecore.EObject;
import org.eclipse.emf.ecore.InternalEObject;

import org.eclipse.emf.ecore.impl.EObjectImpl;

/**
 * @generated
 */
public class GPS_INFO_R extends EObjectImpl implements IGPS_INFO_R {
	protected double latitude = 0.0;
	protected double longitude = 0.0;
	protected double altitude = 0.0;
	protected double positionTime = 0.0;
	protected float hDOP = 0.0f;
	protected float pDOP = 0.0f;
	protected int numberOfSattelites = 0;

	protected GPS_INFO_R() {
		super();
	}

	@Override
	protected EClass eStaticClass() {
		return FcalPackage.Literals.GPS_INFO_R;
	}

	public double getLatitude() {
		return latitude;
	}

	public void setLatitude(double newLatitude) {
		latitude = newLatitude;
	}

	public double getLongitude() {
		return longitude;
	}

	public void setLongitude(double newLongitude) {
		longitude = newLongitude;
	}

	public double getAltitude() {
		return altitude;
	}

	public void setAltitude(double newAltitude) {
		altitude = newAltitude;
	}

	public double getPositionTime() {
		return positionTime;
	}

	public void setPositionTime(double newPositionTime) {
		positionTime = newPositionTime;
	}

	public float getHDOP() {
		return hDOP;
	}

	public void setHDOP(float newHDOP) {
		hDOP = newHDOP;
	}

	public float getPDOP() {
		return pDOP;
	}

	public void setPDOP(float newPDOP) {
		pDOP = newPDOP;
	}

	public int getNumberOfSattelites() {
		return numberOfSattelites;
	}

	public void setNumberOfSattelites(int newNumberOfSattelites) {
		numberOfSattelites = newNumberOfSattelites;
	}

	@Override
	public Object eGet(int featureID, boolean resolve, boolean coreType) {
		switch (featureID) {
		case FcalPackage.GPS_INFO_R__LATITUDE:
			return getLatitude();
		case FcalPackage.GPS_INFO_R__LONGITUDE:
			return getLongitude();
		case FcalPackage.GPS_INFO_R__ALTITUDE:
			return getAltitude();
		case FcalPackage.GPS_INFO_R__POSITION_TIME:
			return getPositionTime();
		case FcalPackage.GPS_INFO_R__HDOP:
			return getHDOP();
		case FcalPackage.GPS_INFO_R__PDOP:
			return getPDOP();
		case FcalPackage.GPS_INFO_R__NUMBER_OF_SATTELITES:
			return getNumberOfSattelites();
		}
		return super.eGet(featureID, resolve, coreType);
	}

	@Override
	public void eSet(int featureID, Object newValue) {
		switch (featureID) {
		case FcalPackage.GPS_INFO_R__LATITUDE:
			setLatitude((Double) newValue);
			return;
		case FcalPackage.GPS_INFO_R__LONGITUDE:
			setLongitude((Double) newValue);
			return;
		case FcalPackage.GPS_INFO_R__ALTITUDE:
			setAltitude((Double) newValue);
			return;
		case FcalPackage.GPS_INFO_R__POSITION_TIME:
			setPositionTime((Double) newValue);
			return;
		case FcalPackage.GPS_INFO_R__HDOP:
			setHDOP((Float) newValue);
			return;
		case FcalPackage.GPS_INFO_R__PDOP:
			setPDOP((Float) newValue);
			return;
		case FcalPackage.GPS_INFO_R__NUMBER_OF_SATTELITES:
			setNumberOfSattelites((Integer) newValue);
			return;
		}
		super.eSet(featureID, newValue);
	}

	@Override
	public void eUnset(int featureID) {
		switch (featureID) {
		case FcalPackage.GPS_INFO_R__LATITUDE:
			setLatitude(0.0);
			return;
		case FcalPackage.GPS_INFO_R__LONGITUDE:
			setLongitude(0.0);
			return;
		case FcalPackage.GPS_INFO_R__ALTITUDE:
			setAltitude(0.0);
			return;
		case FcalPackage.GPS_INFO_R__POSITION_TIME:
			setPositionTime(0.0);
			return;
		case FcalPackage.GPS_INFO_R__HDOP:
			setHDOP(0.0f);
			return;
		case FcalPackage.GPS_INFO_R__PDOP:
			setPDOP(0.0f);
			return;
		case FcalPackage.GPS_INFO_R__NUMBER_OF_SATTELITES:
			setNumberOfSattelites(0);
			return;
		}
		super.eUnset(featureID);
	}

	@Override
	public boolean eIsSet(int featureID) {
		switch (featureID) {
		case FcalPackage.GPS_INFO_R__LATITUDE:
			return latitude != 0.0;
		case FcalPackage.GPS_INFO_R__LONGITUDE:
			return longitude != 0.0;
		case FcalPackage.GPS_INFO_R__ALTITUDE:
			return altitude != 0.0;
		case FcalPackage.GPS_INFO_R__POSITION_TIME:
			return positionTime != 0.0;
		case FcalPackage.GPS_INFO_R__HDOP:
			return hDOP != 0.0f;
		case FcalPackage.GPS_INFO_R__PDOP:
			return pDOP != 0.0f;
		case FcalPackage.GPS_INFO_R__NUMBER_OF_SATTELITES:
			return numberOfSattelites != 0;
		}
		return super.eIsSet(featureID);
	}
} //GPS_INFO_R
